docs/.vuepress/theme/components/SchemaStructuredData.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div>
+    <script
+      v-if="meta_structuredData"
+      type="application/ld+json"
+      v-html="meta_structuredData"
+    ></script>
+  </div>
+</template>
+
+<script>
+import * as dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import timezone from 'dayjs/plugin/timezone';
+
+dayjs.extend(utc);
+dayjs.extend(timezone);
+dayjs.tz.setDefault('Poland');
+
+export default {
+  name: 'SchemaStructuredData',
+  computed: {
+    meta_data() {
+      if (!this.$page) {
+        return;
+      }
+
+      const organization = this.$site.themeConfig.organization;
+
+      return {
+        title: this.$page.title
+          ? this.$page.title.toString().replace(/["|'|\\]/g, '')
+          : null,
+        description: this.$page.frontmatter.description
+          ? this.$page.frontmatter.description
+            .toString()
+            .replace(/["|'|\\]/g, '')
+          : null,
+        image: organization.image,
+        published:
+          dayjs(this.$page.frontmatter.date).toISOString() ||
+          dayjs(this.$page.lastUpdated).toISOString(),
+        modified: dayjs(this.$page.lastUpdated).toISOString(),
+      };
+    },
+    meta_canonicalUrl() {
+      if (
+        !this.$page.frontmatter.canonicalUrl ||
+        !this.$page.path ||
+        !this.$page.hostname
+      ) {
+        return null;
+      }
+
+      return this.$page.frontmatter.canonicalUrl
+        ? this.$page.frontmatter.canonicalUrl
+        : `${this.$page.hostname}/docs${this.$page.path}`;
+    },
+
+    // Inject Schema.org structured data
+    meta_structuredData() {
+      const organization = this.$site.themeConfig.organization;
+      const structuredData = {
+        '@context': 'https://schema.org/',
+        '@type': 'WebPage',
+        name: this.meta_data.title || null,
```

Comment:
Is it possible to have name, headline and description to be a bit more descriptive like the original issue example json ld